boards/imix/src/power.rs

```diff
@@ -0,0 +1,106 @@
+extern crate kernel;
+extern crate sam4l;
+
+use kernel::hil::Controller;
+use sam4l::gpio::{PA, PB, PC};
+use sam4l::gpio::GPIOPin;
+use sam4l::gpio::PeripheralFunction;
+use sam4l::gpio::PeripheralFunction::{A, B};
+
+type DetachablePin = (&'static GPIOPin, Option<PeripheralFunction>);
+
+trait Detachable {
+    fn detach(&self);
+    fn restore(&self, function: Option<PeripheralFunction>);
+}
+
+impl Detachable for GPIOPin {
+    fn detach(&self) {
+        self.configure(None);
+        self.enable_output();
+        self.clear();
+    }
+
+    fn restore(&self, function: Option<PeripheralFunction>) {
+        self.configure(function);
+    }
+}
+
+trait PowerGated {
+    fn power(&self, state: bool);
+}
+
+struct ImixSubmodule {
```

Comment:
This is a taste thing, but since this type is already namespaced under the `imix` crate, I don't think it's necessary to prefix it's name with `Imix`.